src/main/java/io/pivotal/cfapp/controller/LegacyWorkloadsController.java

```diff
@@ -28,16 +36,24 @@ public LegacyWorkloadsController(
         this.util = new TkServiceUtil(tkService);
     }
 
+
     @GetMapping(value = { "/snapshot/detail/legacy" } )
-	public Mono<ResponseEntity<Workloads>> getLegacyWorkloads(@RequestParam("stacks") String stacks) {
+    public Mono<ResponseEntity<Workloads>> getLegacyWorkloads(@RequestParam(value = "stacks", defaultValue = "", required = false) String stacks,
+    @RequestParam(value = "serviceOfferings", defaultValue = "", required = false) String serviceOfferings
+    ) {
         final WorkloadsBuilder builder = Workloads.builder();
+        final WorkloadsFilter workloadFilters = WorkloadsFilter.builder()
+        .stacks(Set.copyOf(Arrays.asList(stacks.split("\\s*,\\s*"))))
+        .serviceOfferings(Set.copyOf(Arrays.asList(serviceOfferings.split("\\s*,\\s*"))))
+        .build();
         return service
-                .getLegacyApplications(stacks)
+                .getLegacyApplications(workloadFilters)
```

Comment:
Local variable name should match type name so please rename to `workloadsFilter`.